### PR TITLE
Add purge and chef-node-name options to server delete.

### PR DESCRIPTION
**Pull request by jcobol**
Add option to server delete to also delete from Chef Server.

### CODE REVIEW

File: lib/chef/knife/bmcs_server_delete.rb [hunk 1]
```diff
@@ -29,16 +29,38 @@ class BmcsServerDelete < Knife
              long: '--wait SECONDS',
              description: 'Wait for the instance to be terminated. 0=infinite'
 
+      option :prune,
+             long: '--prune',
```
Comment (briangustafson) [inline lib/chef/knife/bmcs_server_delete.rb]: --purge would be more consistent with other knife commands.

File: lib/chef/knife/bmcs_server_delete.rb [hunk 2]
```diff
@@ -29,16 +29,38 @@ class BmcsServerDelete < Knife
...
+             description: 'Also remove node from Chef server.'
```
Comment (briangustafson) [inline lib/chef/knife/bmcs_server_delete.rb]: Should mention here that the display name is used unless --chef-node-name is specified.

File: lib/chef/knife/bmcs_server_delete.rb [hunk 3]
```diff
@@ -29,16 +29,38 @@ class BmcsServerDelete < Knife
...
+
+      option :chef_node_name,
+             long: '--chef-node-name NAME',
```
Comment (briangustafson) [inline lib/chef/knife/bmcs_server_delete.rb]: I would mirror the definition from server create: 
      option :chef_node_name,
             short: '-N NAME',
             long: '--node-name NAME',

File: lib/chef/knife/bmcs_server_delete.rb [hunk 4]
```diff
@@ -29,16 +29,38 @@ class BmcsServerDelete < Knife
...
+        response = check_can_access_instance(config[:instance_id])
+
+        ui.msg "Instance name: #{response.data.display_name}"
+        deletion_prompt = 'Delete server? (y/n)'
+        chef_node = nil
+        if config[:prune]
+          deletion_prompt = 'Delete server and chef node? (y/n)'
+          node_name = response.data.display_name
+          node_name = config[:chef_node_name] if config[:chef_node_name]
+          chef_node = get_chef_node(node_name)
```
Comment (briangustafson) [inline lib/chef/knife/bmcs_server_delete.rb]: What happens if a node with that name is not found? I would think that we should show an error that includes the name we were looking for.

Comment (jcobol) [inline lib/chef/knife/bmcs_server_delete.rb]: Here is the output in that case:
[vagrant@localhost knife-bmcs]$ knife bmcs server delete ... --prune --chef-node-name doesnotexist
Instance name: johntest
ERROR: The object you are looking for could not be found
Response: node 'doesnotexist' not found

File: spec/integration/server_create_integ_spec.rb [hunk 5]
```diff
@@ -56,34 +75,61 @@ def validate_output(shell, params)
...
-    validate_output(shell, params)
-  end
-
   it 'can create an Oracle Linux instance with all params' do
     params = min_params.merge(extra_params)
     params['--image-id'] = oracle_linux_image_id
     shell = run_server_create(params, 'test_oracle_linux_all_params')
     validate_output(shell, params)
     expect(shell.stdout).to include(params['--display-name'])
+    unless @latest_output.nil?
```
Comment (briangustafson) [inline spec/integration/server_create_integ_spec.rb]: A bit of a nitpick, but this block is repeated 3 times. Seems worth moving into a method.

File: test_output/help/server_delete.txt [hunk 6]
```diff
@@ -3,6 +3,7 @@ knife bmcs server delete --help
 knife bmcs server delete (options)
         --bmcs-config-file FILE      The path to the Oracle BMCS config file. Default: ~/.oraclebmc/config
         --bmcs-profile PROFILE       The profile to load from the Oracle BMCS config file. Default: DEFAULT
+    -N, --node-name NAME             The Chef node name being removed. If not specified, the instance display name will be used.
```
Comment (briangustafson) [inline test_output/help/server_delete.txt]: The name of the Chef node to be removed when using the purge option. If not specified, the instance display name will be used.

File: test_output/help/server_delete.txt [hunk 7]
```diff
@@ -22,6 +23,7 @@ knife bmcs server delete (options)
     -z, --local-mode                 Point knife commands at local repository instead of server
     -u, --user USER                  API Client Username
         --print-after                Show the data after a destructive operation
+        --purge                      Also remove node from Chef server. Chef node name defaults to the instance display name unless node-name is specified.
```
Comment (briangustafson) [inline test_output/help/server_delete.txt]: Remove the corresponding node from the Chef Server. The instance display name will be used as the node name, unless --node-name is specified.

**Pull request merged**